app/Http/Controllers/Api/PostsController.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace Rogue\Http\Controllers\Api;
+
+use Rogue\Models\Signup;
+use Illuminate\Http\Request;
+use Rogue\Services\PostService;
+use Rogue\Repositories\SignupRepository;
+use Rogue\Http\Transformers\PhotoTransformer;
+
+class PostsController extends ApiController
+{
+    /**
+     * The photo service instance.
+     *
+     * @var Rogue\Services\PostService
+     */
+    protected $posts;
+
+    /**
+     * The signup repository instance.
+     *
+     * @var Rogue\Repositories\SignupRepository
+     */
+    protected $signups;
+
+    /**
+     * @var \League\Fractal\TransformerAbstract;
+     */
+    protected $transformer;
+
+    /**
+     * Create a controller instance.
+     *
+     * @param  PostContract  $posts
+     * @return void
+     */
+    public function __construct(PostService $posts, SignupRepository $signups)
+    {
+        $this->posts = $posts;
+        $this->signups = $signups;
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        $transactionId = incrementTransactionId($request);
```

Comment:
Ah i love that idea. I might hold off on that as a separate issue since it would be a little out of the scope of this work. But i will document that.